core/src/main/java/co/aikar/commands/BaseCommand.java

```diff
@@ -64,69 +65,228 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+/**
+ * BaseCommand stores data about a single command tree.
+ * This includes {@link CommandManager who} manages it, what {@link RegisteredCommand subcommands} it has and so on.
```

Comment:
Per comments below. Should be noted that base commands are more so a "Command Group", that shares common elements (subcommand paths, config, intent, permissions, etc)

base command is a command group of registered commands.